ledger/db/src/test_utils/mod.rs

```diff
@@ -429,7 +429,13 @@ pub fn add_block_contents_to_ledger(
     let new_block = if num_blocks > 0 {
         let parent = ledger_db.get_block(num_blocks - 1)?;
 
-        Block::new_with_parent(block_version, &parent, &Default::default(), &block_contents)
+        Block::new_with_parent(
+            block_version,
+            &parent,
+            &Default::default(),
+            &block_contents,
+            parent.timestamp + 1,
```

Comment:
Not sure if this causes any problems (or will cause) but this will set a timestamp even for block versions that do not support it. Perhaps worth setting to 0 based on the block_version?

Reply:
Let me know what you think, I pushed the logic into `Block::new()` as well as `api/src/convert/block.rs`

Reply:
I think having that particular function have subtle behavior were it zeros out an unexpected timestamp is not ideal since that could mask some misbehavior up the chain that we didn't expect, where somehow a timestamp appeared where it shouldn't have, so as I suggested in the other comment I think panicing there would be my preference. But open for other ideas/suggestions.